src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs

```diff
@@ -55,57 +61,202 @@ internal static partial class JsonWriterHelper
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool NeedsEscaping(char value) => value > LastAsciiCharacter || AllowList[value] == 0;
 
-        public static int NeedsEscaping(ReadOnlySpan<byte> value, JavaScriptEncoder encoder)
+#if BUILDING_INBOX_LIBRARY
+        private static readonly Vector128<short> s_mask_UInt16_0x20 = Vector128.Create((short)0x20); // Space ' '
+
+        private static readonly Vector128<short> s_mask_UInt16_0x22 = Vector128.Create((short)0x22); // Quotation Mark '"'
```

Comment:
We need to use `short` since that's what the intrinsics I am using support (such as `Sse2.CompareEqual`).

Negative is fine here, since that would get caught be the `< 0x20` check and there is no char/ushort value that circles back around to the same sign when casted as short.

The range 32768 (`short.MaxValue + 1`) to 65535 (`char.MaxValue`) all fits into the negative range for a short (from -1 to -32768).

> for a surrogate char I believe the next char is always < 0x8000 and > 0xFF

I don't know what you mean. Surrogate chars have to be in high-low pairs (for valid text) where the high-surrogate followed by a low-surrogate being 0xD800–0xDBFF and low being 0xDC00–0xDFFF.

>  Do we have tests for this (surrogate char which is >= 0x8000)

I'll add a test for a character >= 0x8000 (but those won't be surrogates, since surrogates require two characters to represent).

Reply:
> We need to use short since that's what the intrinsics I am using support (such as Sse2.CompareEqual).

But there is an overload of CompareEqual for ushort as well.

> Negative is fine here, since that would get caught be the < 0x20 check and there is no char/ushort value that circles back around to the same sign when casted as short.

That doesn't seem intuitive. There is also a comparison for >0x7E

> I don't know what you mean. Surrogate chars have to be in high-low pairs (for valid text) where the high-surrogate followed by a low-surrogate being 0xD800–0xDBFF and low being 0xDC00–0xDFFF.

A low surrogate is always negative w.r.t. signed short.